examples/shouldi/shouldi/cargo_audit.py

```diff
@@ -0,0 +1,41 @@
+import json
+import asyncio
+from typing import Dict, Any
+
+from dffml.df.base import op
+from dffml.df.types import Definition
+
+package_src_dir = Definition(name="package_src_dir", primitive="str")
+cargo_audit_output = Definition(
+    name="golangci_lint_output", primitive="Dict[str, Any]"
+)
+
+
+class Cargo_AuditError(Exception):
+    """
+    Raised when cargo-audit fails
+    """
+
+
+@op(inputs={"pkg": package_src_dir}, outputs={"report": cargo_audit_output})
+async def run_cargo_audit(pkg: str) -> Dict[str, Any]:
+    """
+    CLI usage: dffml service dev run -log debug shouldi.cargo_audit:run_cargo_audit -pkg .
+    """
+    proc = await asyncio.create_subprocess_exec(
+        "cargo",
```

Comment:
```suggestion
        "cargo-audit",
```

Reply:
I think this is the source of your error, since when you split cargo-audit into `cargo audit` you asked the `cargo` command to run the `audit` subcommand, which doesn't exist. We want to run the `cargo-audit` command.

Reply:
@Yash-Varshney, You mentioned that you want to run `cargo-audit audit --json`, if that is the case. You should have 

```python
        "cargo-audit",
        "audit",
        "--json",
```

Just to check your understanding. Can you explain what `asyncio.create_subprocess_exec`  is doing here.

Reply:
@pdxjohnny roughly what I believe, it is executing these commands as we do on our terminal

Reply:
sorry, I should have written those but as it took me a long time to come back to this PR, I forget the two `audits` in it.